exopy/tasks/tasks/logic/loop_geomspace_interface.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright 2015-2023 by Exopy Authors, see AUTHORS for more details.
+#
+# Distributed under the terms of the BSD license.
+#
+# The full license is in the file LICENCE, distributed with this software.
+# -----------------------------------------------------------------------------
+"""Interface allowing to use a linspace in a LoopTask
+
+"""
+import numbers
+from decimal import Decimal
+
+import numpy as np
+from atom.api import Str
+
+from ..task_interface import TaskInterface
+from ..validators import Feval
+
+
+class GeomspaceLoopInterface(TaskInterface):
+    """ Common logic for all loop tasks.
+
+    """
+    #: Value at which to start the loop.
+    start = Str('1.0').tag(pref=True, feval=Feval(types=numbers.Real))
+
+    #: Value at which to stop the loop (included)
+    stop = Str('100.0').tag(pref=True, feval=Feval(types=numbers.Real))
+
+    #: Step between loop values.
+    num = Str('10').tag(pref=True, feval=Feval(types=numbers.Real))
```

Comment:
The validation should enforce we get an integer

Reply:
just updated

Reply:
So I changed it to Str, but then that broke the format_and_eval_string call later on. I set it up as a Str originally because that allows the num variable to be set by a database entry for another task.